test/blocks/store/init_test.clj

```diff
@@ -6,7 +6,25 @@
     [clojure.test :refer :all])
   (:import
     blocks.store.file.FileBlockStore
-    blocks.store.memory.MemoryBlockStore))
+    blocks.store.memory.MemoryBlockStore
+    java.io.File))
+
+
+(defmulti join-path
+  (fn [& parts]
+    (if (= (first parts) File/separator)
+      :absolute
+      :relative)))
```

Comment:
It doesn't really seem like this needs to be a multimethod - nothing outside this namespace should be using/extending this since it's a test utility. It'd be simpler to make this a pair of regular private functions:
```clojure
(defn- relative-path
  [& parts]
  (str/join File/separator parts))

(defn- absolute-path
  [& parts]
  (str File/separator (apply relative-path parts)))
```

Reply:
I fully agree, a multimethod seems like overengineering here.